apis/externalsecrets/v1beta1/externalsecret_types.go

```diff
@@ -110,13 +109,35 @@ const (
 	TemplateEngineV2 TemplateEngineVersion = "v2"
 )
 
-// +kubebuilder:validation:MinProperties=1
-// +kubebuilder:validation:MaxProperties=1
 type TemplateFrom struct {
 	ConfigMap *TemplateRef `json:"configMap,omitempty"`
 	Secret    *TemplateRef `json:"secret,omitempty"`
+	// +optional
+	// +kubebuilder:default="Values"
+	Scope TemplateScope `json:"scope,omitempty"`
```

Comment:
I think we should be able to remove the  `scope` . Annotations, labels and secret keys do not allow template literal characters `{{ foo }}`.

Reply:
The scope is important because it would make a change to how we interpret the map for annotations and labels. This allows setting GetAllSecrets contents to keys in Annotations (note that the keys themselves can be templated this way)

Reply:
I see now, this is needed to allow factoring out the `templateFrom[].literal` into a  `configMap` or `secret` so it can be reused.
TBH i struggled really hard to understand what that `scope` actually means and why i would need it. So essentially, with `literal|configMap|secret` we point to a string and the `scope` defines how it should be interpreted: either it should unmarshal into a `map[string]string` or a `string`.

I think we can make it less complex (from maintainability and UX perspective) if we push the template execution down to the configMap / secret level. I called it `parsePolicy` here - just for the sake of giving it a name :smile: 
This way we can get rid of the `scope` alltogether and give the user the ability to selectively change the interpretation.

```yaml
spec:
  refreshInterval: "10s"
  secretStoreRef:
    name: fake
    kind: ClusterSecretStore
  target:
    name: "fart"
    template:
      templateFrom:
        # ex 1
        - configMap:
            name: "my-app-config"
            items:
            - key: "config.yaml"
              parsePolicy: String # default
        # ex 2
        - configMap:
            name: "my-generic"
            items:
            - key: "generated"
              parsePolicy: Map # executes the template and unmarshals it as map[string]string
        # ex 3
        - literal: |
            {{- $obj := .foo_bar_v1 | b64dec | fromYaml -}}
            {{- range $key, $val := .foo_bar_v1 | b64dec | fromYaml }}
            {{ $key }}: "{{ $val }}"
            {{- end  }}
            other: "stuff"
```

I'm sure we'll find better names than that :point_up_2:, that's just to illustrate my thoughts. What do you think?

Reply:
Just added some refactoring to try to take away the complexity. IMO things looks better now. I changed the name Scope to `templateAs` still with `Values` or `KeysAndValues` option, so users can have a better experience when using the new feature

Reply:
Thanks, yeah that looks good now, also way cleaner.